src/applications/oscilloscope.py

```diff
@@ -1,117 +1,247 @@
+
         self.line.set_ydata(self.ydata)
-        return self.line,
 
+        if self.show_rolling_mean:
+            new_rm_value = self.update_rolling_mean()
+            self.rolling_mean_line.set_ydata(self.rolling_mean)
 
+        if self.show_reading_text:
+            text = self._get_text_value(y)
+            if self.show_rolling_mean:
+                text = f'{text}\n {self._get_text_value(new_rm_value)}'
+                measured_stdev = self.get_new_rolling_stdev_val()
+                expected_stdev = np.sqrt(new_rm_value)
+                text = f'{text}\n {self._get_text_value(measured_stdev)} ({self._get_text_value(expected_stdev)})'
+            self.current_value_text.set_text(text)
 
-class MainApplicationView():
-    def __init__(self, main_frame):
+        return self.line,
+
+    def get_new_rolling_mean_val(self):
+        """ Calculates new rolling mean value. """
+        values_of_interest = list(self.ydata)[len(self.ydata) - 2 * self.half_window_size:
+                                              len(self.ydata)]
+        return np.mean(values_of_interest)
+
+    def get_new_rolling_stdev_val(self):
+        """ Calculates new rolling standard deviation. """
+        values_of_interest = list(self.ydata)[len(self.ydata) - 2 * self.half_window_size:
+                                              len(self.ydata)]
+        return np.std(values_of_interest)
+
+    def update_rolling_mean(self):
+        """ Updates internal rolling mean buffer. """
+        rm = collections.deque(self.rolling_mean[self.half_window_size: len(self.ydata) - self.half_window_size])
+        rm.popleft()
+        new_rm_value = self.get_new_rolling_mean_val()
+        rm.append(new_rm_value)
+        self.rolling_mean[self.half_window_size: len(self.ydata) - self.half_window_size] = list(rm)
+        return new_rm_value
+
+    def get_rolling_mean(self):
+        """ Computes rolling mean of current data. """
+        rolling_mean = get_rolling_mean(self.ydata, self.rolling_mean_window_size)
+        rolling_mean[:self.half_window_size] = np.nan
+        rolling_mean[len(self.ydata) - self.half_window_size:] = np.nan
+        return rolling_mean
+
+    @staticmethod
+    def _get_text_value(value: float):
+        """ Formats number for display in figure text. """
+        rounded_value = np.around(value, 1)
+        return f'{rounded_value:,}'
+
+
+class MainApplicationView:
+    """
+    Represents the application view.
+
+    Contains a matplotlib figure canvas and a sidebar.
+    """
+
+    def __init__(self, main_frame: Tk.Tk):
         frame = Tk.Frame(main_frame)
         frame.pack(side=Tk.LEFT, fill=Tk.BOTH, expand=True)
 
-        self.scope_view = ScopeFigure(args.scope_width)
+        rolling_mean_window_size = np.ceil(args.rolling_mean_window / args.animation_update_interval)
+        self.scope_view = ScopeFigure(args.scope_width, int(rolling_mean_window_size))
```

Comment:
You don't instantiate ScopeFigure with all the arguments. There's currently no way to turn  off the current value / mean / std display.